rostime/include/ros/impl/duration.h

```diff
@@ -60,6 +63,9 @@ namespace ros {
   template<class T>
   T& DurationBase<T>::fromNSec(int64_t t)
   {
+    if ((t / 1000000000) < INT_MIN || (t / 1000000000) > INT_MAX)
```

Comment:
Maybe adding a local variable to store the 64 bit values would make sense to avoid repeating the computation.

Same below.